xarray/tests/test_dataset.py

```diff
@@ -5051,9 +5051,9 @@ def test_dropna(self) -> None:
         ):
             ds.dropna("foo")
         with pytest.raises(ValueError, match=r"invalid how"):
-            ds.dropna("a", how="somehow")  # type: ignore
+            ds.dropna("a", how="somehow")
```

Comment:
Not entirely sure if/ how these are related with the PR.

Reply:
Unfortunately my guess is that `@_deprecate_positional_args` doesn't pass types through, so these come as `Any`, so will need to be re-enabled when we remove the decorator...

Reply:
I thought `@functools.wraps` conserves the `__signature__` of the wrapped function...

Reply:
what it does (among other things) is set `wrapper.__wrapped__ = wrapped`. According to [this article]
(https://hynek.me/articles/decorators/) (which admittedly is a bit old), that doesn't mean the signature is the wrapped one's, just that `inspect.signature` is smart enough to display the wrapped signature.

In other words, it depends on how `mypy` fetches the type annotations.

Reply:
I think it should be easy to add typing to the deprecation helper. It should return `Callable[T, T]`, where `T = TypeVar("T", bound=Callable)`.

This means that mypy will complain if someone still uses positional arguments, which I think is exactly what we want.

Reply:
I tried but feel free to push to the PR if it still need fixes.

Reply:
I'm mobile only currently, but I can see if I can fix it in the next two days.

Reply:
I think I got it now (maybe). But that means libraries that type check don't get a deprecation period (also fine by me, but then diverging from the runtime behaviour).